File: fix_pdf.py
import re

with open('frontend/app/results/[id]/page.jsx', 'r', encoding='utf-8') as f:
    page = f.read()

# Replace downloadPDF function
pdf_old = r"const downloadPDF = async \(\) => \{[\s\S]*?toast\.error\(`PDF Failed: \$\{err\.message \|\| 'Check console'\}`, \{ id: toastId \}\);\s*\}\s*\};"
pdf_new = """const downloadPDF = () => {
    toast.info('Opening Print Dialog. Save as PDF.');
    setTimeout(() => window.print(), 500);
  };"""

page = re.sub(pdf_old, pdf_new, page)

# Wrap buttons in print:hidden
buttons_old = """<div className="mt-8 flex flex-col md:flex-row gap-3 max-w-[400px] ml-auto">"""
buttons_new = """<div className="mt-8 flex flex-col md:flex-row gap-3 max-w-[400px] ml-auto print:hidden">"""
page = page.replace(buttons_old, buttons_new)

# Ensure Navbar has print:hidden in the page just in case
page = page.replace("<NavBar />", "<div className=\"print:hidden\"><NavBar /></div>")

with open('frontend/app/results/[id]/page.jsx', 'w', encoding='utf-8') as f:
    f.write(page)
print("Updated PDF logic in page.jsx")

File: fix_pdf.js
const fs = require('fs');
let js = fs.readFileSync('frontend/app/results/[id]/page.jsx', 'utf8');

// 1. Add id to container
js = js.replace(
  `<div className="max-w-[1200px] mx-auto px-6 py-12">`,
  `<div id="pdf-content" className="max-w-[1200px] mx-auto px-6 py-12">`
);

// 2. Replace downloadPDF
const oldFn = `  const downloadPDF = () => {
    toast.info('Opening Print Dialog. Save as PDF.');
    setTimeout(() => window.print(), 500);
  };`;

const newFn = `  const downloadPDF = async () => {
    toast.loading('Generating Official Notice PDF...', { id: 'pdf' });
    try {
      const html2pdf = (await import('html2pdf.js')).default;
      const element = document.getElementById('pdf-content');
      
      const opt = {
        margin:       10,
        filename:     \`Legal_Notice_\${report.id.substring(0,8)}.pdf\`,
        image:        { type: 'jpeg', quality: 0.98 },
        html2canvas:  { scale: 2, useCORS: true, backgroundColor: '#09090b' },
        jsPDF:        { unit: 'mm', format: 'a4', orientation: 'portrait' }
      };

      await html2pdf().set(opt).from(element).save();
      toast.success('PDF Downloaded', { id: 'pdf' });
    } catch (err) {
      console.error(err);
      toast.error('Failed to generate PDF', { id: 'pdf' });
    }
  };`;

js = js.replace(oldFn, newFn);
fs.writeFileSync('frontend/app/results/[id]/page.jsx', js);
console.log("PDF generation added");
